### PR TITLE
client/web: fix back button on devices with URL prefix

### PR DESCRIPTION
**Pull request by willnorris**
Move Header component inside Router so that links are relative to the router base URL.

Updates tailscale/corp#14335

### CODE REVIEW

File: client/web/src/components/app.tsx [hunk 1]
```diff
@@ -112,9 +113,9 @@ function Header({
         </div>
         <LoginToggle node={node} auth={auth} newSession={newSession} />
       </div>
-      {loc !== "/" && (
+      {loc !== homeURL && loc !== homeURL + "/" && (
```
Comment (soniaappasamy) [inline client/web/src/components/app.tsx:116]: Oh does this not pick up on the base we set?

**Review by soniaappasamy — APPROVED** [review]

File: client/web/src/components/app.tsx (hunk 1 above)
Comment (willnorris) [reply client/web/src/components/app.tsx:116]: apparently not.  On my Synology, it was linking me back to the root without the base

Comment (willnorris) [reply client/web/src/components/app.tsx:116]: Ah, I think it's because our <Header> is outside our <Router>.  Lemme try moving that.

Comment (willnorris) [reply client/web/src/components/app.tsx]: there we go... that's much better.

**Pull request merged**